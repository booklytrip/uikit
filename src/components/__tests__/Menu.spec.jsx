import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MenuItem, Menu } from '../Menu';

describe('MenuItem', () => {
    it('renders without crashing', () => {
        mount(<MenuItem />);
    });

    it('should call onClick event with arguments if clicked', () => {
        const spy = jest.fn();
        const item = mount(
            <MenuItem
                value="value"
                label="label"
                onClick={ spy }
            />
        );
        item.simulate('click');

        expect(spy).toHaveBeenCalledWith('value', 'label');
    });

    describe('Snapshots', () => {
        it('renders with label', () => {
            const rendered = renderer.create(<MenuItem label="Label" />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders in active state', () => {
            const rendered = renderer.create(<MenuItem active />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});

describe('Menu', () => {
    const items = [
        { value: 1, label: 'First' },
        { value: 2, label: 'Second' },
        { value: 3, label: 'Third' },
    ];

    it('renders without crashing', () => {
        mount(<Menu />);
    });

    it('should call onClick event with arguments if clicked on item', () => {
        const spy = jest.fn();
        const menu = mount(
            <Menu items={ items } onClick={ spy } />
        );
        menu.find(MenuItem).at(1).simulate('click');

        expect(spy).toHaveBeenCalledWith(2, 'Second');
    });

    describe('Snapshots', () => {
        it('renders with items', () => {
            const rendered = renderer.create(<Menu items={ items } />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders with active item', () => {
            const rendered = renderer.create(<Menu items={ items } active={ 2 } />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders vertically', () => {
            const rendered = renderer.create(<Menu items={ items } vertically />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders secondary', () => {
            const rendered = renderer.create(<Menu items={ items } secondary />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders tabular', () => {
            const rendered = renderer.create(<Menu items={ items } tabular />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders attached', () => {
            const rendered = renderer.create(<Menu items={ items } attached="top" />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders pointing', () => {
            const rendered = renderer.create(<Menu items={ items } pointing />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
