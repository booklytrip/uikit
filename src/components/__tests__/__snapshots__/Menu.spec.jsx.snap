exports[`Menu Snapshots renders attached 1`] = `
<div
  className="ui menu top attached">
  <a
    className="item"
    onClick={[Function]}>
    First
  </a>
  <a
    className="item"
    onClick={[Function]}>
    Second
  </a>
  <a
    className="item"
    onClick={[Function]}>
    Third
  </a>
</div>
`;

exports[`Menu Snapshots renders pointing 1`] = `
<div
  className="ui menu pointing">
  <a
    className="item"
    onClick={[Function]}>
    First
  </a>
  <a
    className="item"
    onClick={[Function]}>
    Second
  </a>
  <a
    className="item"
    onClick={[Function]}>
    Third
  </a>
</div>
`;

exports[`Menu Snapshots renders secondary 1`] = `
<div
  className="ui menu secondary">
  <a
    className="item"
    onClick={[Function]}>
    First
  </a>
  <a
    className="item"
    onClick={[Function]}>
    Second
  </a>
  <a
    className="item"
    onClick={[Function]}>
    Third
  </a>
</div>
`;

exports[`Menu Snapshots renders tabular 1`] = `
<div
  className="ui menu tabular">
  <a
    className="item"
    onClick={[Function]}>
    First
  </a>
  <a
    className="item"
    onClick={[Function]}>
    Second
  </a>
  <a
    className="item"
    onClick={[Function]}>
    Third
  </a>
</div>
`;

exports[`Menu Snapshots renders vertically 1`] = `
<div
  className="ui menu">
  <a
    className="item"
    onClick={[Function]}>
    First
  </a>
  <a
    className="item"
    onClick={[Function]}>
    Second
  </a>
  <a
    className="item"
    onClick={[Function]}>
    Third
  </a>
</div>
`;

exports[`Menu Snapshots renders with active item 1`] = `
<div
  className="ui menu">
  <a
    className="item"
    onClick={[Function]}>
    First
  </a>
  <a
    className="item active"
    onClick={[Function]}>
    Second
  </a>
  <a
    className="item"
    onClick={[Function]}>
    Third
  </a>
</div>
`;

exports[`Menu Snapshots renders with items 1`] = `
<div
  className="ui menu">
  <a
    className="item"
    onClick={[Function]}>
    First
  </a>
  <a
    className="item"
    onClick={[Function]}>
    Second
  </a>
  <a
    className="item"
    onClick={[Function]}>
    Third
  </a>
</div>
`;

exports[`MenuItem Snapshots renders in active state 1`] = `
<a
  className="item active"
  onClick={[Function]} />
`;

exports[`MenuItem Snapshots renders with label 1`] = `
<a
  className="item"
  onClick={[Function]}>
  Label
</a>
`;
